cmake/FindOVS.cmake

```diff
@@ -0,0 +1,131 @@
+# FindOVS.cmake - find Open vSwitch package.
+#
+# Copyright 2022-2023 Intel Corporation
+# SPDX-License-Identifier: Apache 2.0
+#
+
+#-----------------------------------------------------------------------
+# Use pkg-config to search for the modules.
+#-----------------------------------------------------------------------
+find_package(PkgConfig)
+pkg_check_modules(PC_OVS libopenvswitch)
+
+#-----------------------------------------------------------------------
+# Find libraries and include directories.
+#-----------------------------------------------------------------------
+find_path(OVS_INCLUDE_DIR
+    NAMES "openvswitch/version.h"
+    PATHS ${PC_OVS_INCLUDE_DIRS}
+)
+
+find_library(OVS_LIBRARY
+    NAMES openvswitch
+    PATHS ${PC_OVS_LIBRARY_DIRS}
+)
+
+find_library(OFPROTO_LIBRARY
+    NAMES ofproto
+    PATHS ${PC_OVS_LIBRARY_DIRS}
+)
+
+find_library(OVSDB_LIBRARY
+    NAMES ovsdb
+    PATHS ${PC_OVS_LIBRARY_DIRS}
+)
+
+find_library(SFLOW_LIBRARY
+    NAMES sflow
+    PATHS ${PC_OVS_LIBRARY_DIRS}
+)
+
+find_library(LIBVSWITCHD
+    NAMES vswitchd
+    PATHS ${PC_OVS_LIBRARY_DIRS}
+)
+
+find_library(LIBTESTCONTROLLER
+    NAMES testcontroller
+    PATHS ${PC_OVS_LIBRARY_DIRS}
+)
+
+#-----------------------------------------------------------------------
+# Get version number
+#-----------------------------------------------------------------------
+if(PC_OVS_VERSION)
+    set(OVS_VERSION ${PC_OVS_VERSION} CACHE STRING "OVS version")
+elseif(EXISTS "${OVS_INCLUDE_DIR}/openvswitch/version.h")
+    # Extract version string from version.h file.
+    file(STRINGS
+        "${OVS_INCLUDE_DIR}/openvswitch/version.h"
+        _version_string REGEX "OVS_PACKAGE_VERSION")
+    string(REGEX REPLACE
+        "[^0-9.]+([0-9.]+).*" "\\1" OVS_VERSION ${_version_string})
+    set(OVS_VERSION ${PC_OVS_VERSION} CACHE STRING "OVS version")
+    unset(_version_string)
+endif()
+
+#-----------------------------------------------------------------------
+# Handle REQUIRED and QUIET arguments
+#-----------------------------------------------------------------------
+find_package_handle_standard_args(OVS
```

Comment:
Do we need a `REASON_FAILURE_MESSAGE` message here to indicate if one of the `REQUIRED_VARS` is missing? (not sure what the default messaging looks like, but should not fail silently/cryptic message)

Reply:
Good idea. I should give this a try.

Reply:
Ah. `REASON_FAILURE_MESSAGE` isn't supported until CMake 3.16. Our canonical minimum version is 3.15.

`FAIL_MESSAGE` (the related option) is "not recommended" (which doesn't stop eight of the standard find-modules in cmake-3.16 from using it).

Reply:
I did contrive to have `find_package_handle_standard_args()` fail in one of the tests I did during development. IIRC, its default error messages was somewhat cryptic.

Reply:
ok. Can look into this as future improvement if we want to merge this PR now